CodeTracker/CodeTrackerUnitTest/CodingTrackerInputTest.cs

```diff
@@ -0,0 +1,102 @@
+using CodeTracker;
+using Spectre.Console.Testing;
+namespace CodeTrackerUnitTest;
+
+[TestClass]
+public class CodingTrackerInputTest
+{
+    private readonly TestConsole _console;
+    private readonly Input _input;
+    public CodingTrackerInputTest()
+    {
+        _console = new TestConsole();
+        _input = new Input(_console);
+    }
+    [TestMethod]
+    [DataRow(" ")]
+    [DataRow("description")]
+    public void TestingDescriptionInput(string input)
+    {
+        // Arrange
+        if (!string.IsNullOrWhiteSpace(input))
+        {
+            _console.Input.PushTextWithEnter(input);  // Valid input case
+            var result = _input.GetDescription();
+            Assert.AreEqual(input, result);  // Verify the description matches the input
+        }
+        else
+        {
+            _console.Input.PushTextWithEnter(input);  // Invalid input case
+
+            // Act & Assert: Expect an exception due to invalid input
+            Assert.ThrowsException<InvalidOperationException>(() => _input.GetDescription());
+        }
+    }
+
+    [TestMethod]
+    [DataRow("test")]
+    [DataRow("")]
+    [DataRow("22-10-2023")]
+    public void TestDateInput(string input)
+    {
+        var reader = new StringReader(input);
+        Console.SetIn(reader);
+        DateTime result = _input.GetDay();
+        if (!input.Any(char.IsDigit))
+        {
+            Assert.AreEqual(DateTime.Today,result);
+            return;
+        }
+        Assert.AreEqual(string.IsNullOrEmpty(input) ? DateTime.Today : new DateTime(2023,10,22), result);
+    }
+    [TestMethod]
+    [DataRow("20:30")]
+    public void TestStartTimeInput(string input)
+    {
+        var reader = new StringReader(input);
+        Console.SetIn(reader);
+        
+        DateTime result = _input.GetStartTime(new DateTime(2023,10,22));
+        Assert.AreEqual(new DateTime(2023,10,22,20,30,00), result);
+    }
+    [TestMethod]
+    [DataRow("20-10-2023")]
+    [DataRow("")]
+    public void TestOptionalDateInput(string input)
```

Comment:
🟠 This method uses `DateTime.MinValue` for empty input, which may not be intuitive for a beginner.

💡 Clarify this behavior or change it to return null (if nullable types are allowed), making the API more explicit about the absence of a date.

🔧 Add a test case for invalid date input (e.g., invalid-date-string), which should throw an exception.